Removes OpenStack-related information as unsupported for v3 alpha

emanic opened the pull request:
## Description

- Replaces https://github.com/projectcalico/calico/pull/1247
- Removes OpenStack-related information from the v3.0 directory as we do not support the platform for the alpha.


## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

<!-- Writing a release note:
- By default, no release note action is required.
- If you're unsure whether or not your PR needs a note, ask your reviewer for guidance.
- If this PR requires a release note, update the block below to include a concise note describing
  the change and any important impacts this PR may have.
-->

```release-note
None required
```

neiljerram commented on master/usage/routereflector/calico-routereflector.md:
```diff
@@ -18,7 +18,7 @@ However, it may provide a useful framework for future development.
 calico/node container image.
 
-For an OpenStack deployment, read [Configuring BIRD as a BGP Route Reflector](bird-rr-config).
+<!---For an OpenStack deployment, read [Configuring BIRD as a BGP Route Reflector](bird-rr-config).-->
```
I would prefer to delete this than to comment.  Deletion is neater for the current master doc, and we can still get the content back again from git.

neiljerram commented on master/usage/troubleshooting/faq.md:
```diff
@@ -127,10 +127,10 @@ running, you've still got full functionality.
 plugin.
 
--   In the case of OpenStack, this means a Neutron database. Our
+<!---   In the case of OpenStack, this means a Neutron database. Our
```
Likewise here.

neiljerram commented on master/usage/troubleshooting/faq.md:
```diff
@@ -210,11 +210,11 @@ update policy: the reachability information does not change. If later
 again, and again the reachability doesn’t have to change.
 
-## "How does Calico interact with the Neutron API?"
+<!-- ## "How does Calico interact with the Neutron API?"-->
```
And here.

emanic commented:
Thanks @neiljerram . I moved the changes so that they are being made in the v3.0 directory instead, now that we have that. That way, as we make new versions off of master, we won't have to recreate the content, when we add the support back in. I am going to hold off making any changes to the license information, as this is a legal thing. Also, the etcdv2, private cloud, and overall doc intro will be handled in separate tickets. Thanks!

neiljerram commented:
@emanic For this PR I don't think I need to do more detailed review.  Bearing in mind that it's pretty much all deletion, I'm happy to approve as long as

- you are happy with the overall shape of what you're deleting (e.g. v3.0 vs master)

- you're happy with whatever 'OpenStack' mentions still remain (which you can easily check for yourself using the invocation that I noted above).

If you let me know when that is, I'll hit the button!  (Now, perhaps?)

(I'm personally unsure about leaving OpenStack content in master at the moment; I think it could be confusing for 'master' to imply OpenStack support when we don't actually have it working.  But I assume you've had that debate with relevant people, and am happy to defer.)

emanic commented:
Hi @neiljerram thanks so much! We are still working out our exact strategy on the directories. For now, we are thinking to keep the deletions confined to the `master` directory. I plan to still tweak this PR slightly (I think I will address the etcdv2 path page in this one after all and maybe run that command that you provided) but if you feel comfortable approving this, that would be a huge help as I know it is much later where you are. :)

emanic commented:
Hi @neiljerram OK I have gone ahead and removed the info from the etcdv2 paths page and reran the `find . -type f -exec grep --color -nH -i openstack {} +` command in the `v3.0` directory. We only now have a handful of references. The intro will be addressed in a different PR, I don't think we should mess with the license file right now and the private cloud info is also something that I have a different ticket to look into. I've also rebased this against the latest master and expect CI to pass. If you feel comfortable approving this, that will expedite its merge. :) Thanks again for your thoughtful review!

neiljerram commented:
Cool, approved - although it appears there is still a CI issue...
